xds/src/main/java/io/grpc/xds/sds/trust/SdsTrustManagerFactory.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2019 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.xds.sds.trust;
+
+import io.envoyproxy.envoy.api.v2.auth.CertificateValidationContext;
+import io.grpc.Internal;
+import io.netty.handler.ssl.util.SimpleTrustManagerFactory;
+import java.io.IOException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.annotation.Nullable;
+import javax.net.ssl.ManagerFactoryParameters;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.TrustManagerFactory;
+import javax.net.ssl.X509ExtendedTrustManager;
+
+@Internal
+public final class SdsTrustManagerFactory extends SimpleTrustManagerFactory {
```

Comment:
we still want javadoc for internal class. it is unclear why it implements netty's class, who will consume this? by how?

Reply:
The SslContextSecretVolumeSecretProvider (in the other PR) will consume this. The Netty class makes it easy to provide a custom trust manager. Hope there is no issue with that.